### PR TITLE
Changed search space of getCompleteCandidates

### PR DESCRIPTION
**Pull request by metodej**
- getCompleteCandidates for group now searches either all perun users or all users who have ues in any of group extSources or who are vo members
- also added tests for getCompleteCandidates (with group parameter)

### CODE REVIEW

File: perun-core/src/main/java/cz/metacentrum/perun/core/bl/VosManagerBl.java [hunk 1]
```diff
@@ -161,14 +161,14 @@
 
 	/**
 	 * Finds MemberCandidates who can join the Group. If the given vo is not null, it searches only
-	 * users who belong to this Vo.
+	 * users who belong to this Vo or who have ues in any of given extSources.
 	 *
 	 * @param sess session
-	 * @param vo vo if is null, users are searched in whole perun, otherwise only members of this vo are used.
+	 * @param vo vo if is null, users are searched in whole perun, otherwise members of this vo are used and users with ues in given extSources.
```
Comment (stavamichal) [inline perun-core/src/main/java/cz/metacentrum/perun/core/bl/VosManagerBl.java:167]: This sentence is hard to understand. What exactly happened if there is not an empty list of ext sources?

File: perun-core/src/main/java/cz/metacentrum/perun/core/blImpl/VosManagerBlImpl.java [hunk 2]
```diff
@@ -527,34 +528,51 @@ public Vo getVoById(PerunSession sess, int id) throws VoNotExistsException {
...
+	 * only RichUsers who has a member inside this Vo or who has ues in any of given ExtSources.</p>
 	 * <p>The RichUsers are returned with attributes of given names.</p>
 	 *
 	 *
 	 * @param sess session
-	 * @param vo virtual organization, users are searched only inside this vo; if is null, then in the whole Perun
+	 * @param vo virtual organization, users are searched inside this vo; if is null, then in the whole Perun
 	 * @param attrNames names of attributes that will be returned
 	 * @param searchString string used to find users
+	 * @param extSources list of extSources to possibly filter RichUsers by
```
Comment (stavamichal) [inline perun-core/src/main/java/cz/metacentrum/perun/core/blImpl/VosManagerBlImpl.java:545]: You do not really filter them. You are just adding new users by the list of ext sources. (if vo is not null)

**Review by stavamichal — CHANGES_REQUESTED** [review]
I think the Javadoc itself should be changed to better describe how changed methods work now.

File: perun-core/src/main/java/cz/metacentrum/perun/core/bl/VosManagerBl.java (hunk 1 above)
Comment (metodej) [reply perun-core/src/main/java/cz/metacentrum/perun/core/bl/VosManagerBl.java]: I've edited the javadoc.

File: perun-core/src/main/java/cz/metacentrum/perun/core/blImpl/VosManagerBlImpl.java (hunk 2 above)
Comment (metodej) [reply perun-core/src/main/java/cz/metacentrum/perun/core/blImpl/VosManagerBlImpl.java]: I've edited the javadoc.

**Review by stavamichal — APPROVED** [review]

**Review by balcirakpeter — APPROVED** [review]

**Pull request merged**